File: Data Structures & Algorithms/palindromic-substrings/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    constructor(){
        this.count = 0;
    }
    expandCenter(left, right, s) {
        while(left >= 0 && right < s.length && s[left] === s[right]) {
            this.count++;
            left--;
            right++;
        }
    }
    countSubstrings(s) {
        for(let i =0 ;i < s.length;i++) {
            this.expandCenter(i, i, s);
            this.expandCenter(i, i+1, s);
        }
    return this.count;

    }
}

File: Data Structures & Algorithms/boats-to-save-people/submission-0.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int count = 0;
        int left = 0;
        int right = people.length - 1;

        while(left <= right) {
            int currSum = people[left] + people[right];
            if(currSum <= limit) {
                left++;
            }
            right--;
            count++;
        }
        return count;
    }
}